src/oracle/middleware.js

```diff
@@ -1,416 +1,81 @@
-/* eslint-disable no-await-in-loop */
-/* eslint-disable @jessie.js/no-nested-await */
 // @ts-nocheck
 /* eslint-disable func-names */
-/* global fetch, process */
 
-/** 
- * IMPORTS
- */
-import { execSwingsetTransaction } from '../lib/chain.js';
-import {
-  makeRpcUtils,
-  boardSlottingMarshaller,
-  networkConfig,
-  makeAgoricNames
-} from '../lib/rpc.js'
-import axios from 'axios';
-import http from 'http';
-import fs from 'fs';
-import { getCurrent } from '../lib/wallet.js';
-import bodyParser from 'body-parser';
-import express from 'express';
-import { validUrl, delay, saveJSONDataToFile, readJSONFile, readVStorage } from './helper.js';
-import {
-  makeFollower,
-  makeLeader,
-} from '@agoric/casting';
-import { iterateReverse } from '@agoric/casting';
+import { 
+  validUrl, 
+  readJSONFile, 
+  initialiseState, 
+  submitNewJob 
+} from '../helpers/utils.js';
+import { 
+  getAllJobs, 
+  queryTable, 
+  updateTable
+} from '../helpers/db.js'
+import { startBridge } from '../helpers/bridge.js'
+import { 
+  queryPrice, 
+  queryRound, 
+  getLatestSubmittedRound 
+} from '../helpers/chain.js'
 
 // get environment variables
 const {
   PORT = '3000',
-  EI_CHAINLINKURL,
   FROM,
-  SUBMIT_RETRIES = 3,
   BLOCK_INTERVAL = '6',
-  SEND_CHECK_INTERVAL = '12',
+  SEND_CHECK_INTERVAL = '45',
```

Comment:
out of curiosity why was this increased?

Reply:
12 seconds (2 blocks) was not enough in the current testnet and the middleware was sending unnecessary requests. We have to see how this will be in mainnet because the reason for this could be the Agoric testnet which we spun up. It only has 1 validator and some specs might be different from mainnet. We will monitor whether it will be better on devnet and hopefully reduce this.

Reply:
ok.